File: backend/app/routers/triage.py
from fastapi import APIRouter
from ..schemas.triage import TriageInput, TriageOutput
from ..services.triage_agent import run_triage

router = APIRouter()


@router.post("", response_model=TriageOutput)
def triage_preview(payload: TriageInput):
    """Run triage on the provided input without creating a work order."""
    return run_triage(payload)

File: frontend/src/api/triage.ts
import client from './client'
import type { TriageInput, TriageOutput } from '../types'

export const runTriage = (data: TriageInput) =>
  client.post<TriageOutput>('/triage', data).then(r => r.data)

export const runAgentTriagePreview = (data: TriageInput) =>
  client.post<TriageOutput>('/agent/triage-preview', data).then(r => r.data)
